lib/dogma/rule/snake_case_filename.ex

```diff
@@ -0,0 +1,63 @@
+use Dogma.RuleBuilder
+
+defrule Dogma.Rule.SnakeCaseFilename do
+  @moduledoc """
+  Checks if files are using the snake_case_format.
+
+  Favors this:
+
+  a_valid_filename.ex
+
+  over these:
+
+  AValidFilename.ex
+  aValidFilename.ex
+  a valid filename.ex
+  a.valid.filename.ex
+  a-valid-filename.ex
+  """
+
+  def test(_, %{path: path}) do
+    {dirname, filename, ext} = deconstruct_path(path)
+
+    desired_filename = filename
```

Comment:
There is already a util module for attempting to detect whether a name is snake_case, use that instead.

If you can demonstrate that this implementation is better you can replace the implementation with this one :)